### ISSUE TITLE
Feature request: Support AKS with Calico on Windows nodes

### ISSUE THREAD

**Issue by olljanat** [projectcalico/calico#6846]
## Current Behavior
Currently it is possible [Bring your own CNI to AKS](https://docs.microsoft.com/en-us/azure/aks/use-byo-cni?tabs=azure-cli) and instruction for that is available in [this](https://projectcalico.docs.tigera.io/getting-started/kubernetes/managed-public-cloud/aks) Calico document. However that same document says that "Windows dataplane is not supported."

## Expected Behavior
It would be nice to have Windows also supported or at least discussed in here that what need to happen that it can be supported or explanation why it cannot be.

## Possible Solution
I already quicky tested that deploying Calico as [HostProcess containers to Windows](https://projectcalico.docs.tigera.io/getting-started/windows-calico/quickstart#install-calico-for-windows-using-hostprocess-containers) partly works if these settings:
```
CNI_BIN_DIR: "c:\\k\\azurecni\\bin"
CNI_CONF_DIR: "c:\\k\\azurecni\\netconf"
CNI_CONF_FILENAME: "10-azure.conflist"
```
are used in `calico-windows.yaml`

As partly I mean that `calico-node.exe` is already running on Windows nodes with that config and pods gets IP addresses from Calico IPAM but Windows nodes are still stuck in `NotReady` status. However I was not able to figure out that why (at least yet).

In additionally I found that this kind of deployment can be used to get shell access to those Windows nodes:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: debug-windows
  labels:
    app: debug-windows
spec:
  replicas: 1
  selector:
    matchLabels:
      app: debug-windows
  template:
    metadata:
      name: debug-windows
      labels:
        app: debug-windows
    spec:
      hostNetwork: true
      securityContext:
        windowsOptions:
          hostProcess: true
          runAsUserName: "NT AUTHORITY\\Local service"
      hostNetwork: true
      containers:
      - name: debug-windows
        image: mcr.microsoft.com/windows/servercore:ltsc2022
        command:
          - ping
          - -t
          - 127.0.0.1
      nodeSelector:
        "kubernetes.io/os": windows
      tolerations:
      - key: "node.cloudprovider.kubernetes.io/uninitialized"
        operator: "Exists"
      - key: "node.kubernetes.io/not-ready"
        operator: "Exists"
```

**Comment by coutinhop** [comment]
@olljanat just a quick update: it seems like there is currently no support for BYO CNI on Windows in AKS (the Azure CNI is installed on Windows nodes regardless of if `--network-plugin none` is used), se we're blocked on AKS adding support. Once that is done, Calico CNI on Windows nodes should just work in AKS
